pkg/release/build.go

```diff
@@ -167,7 +167,7 @@ func MakeRelease(platform Platform, opts BuildOptions, pkgDir string) error {
 		if opts.BuildTag == "" {
 			return errors.Newf("must set BuildTag if Release is set")
 		}
-		buildArgs = append(buildArgs, fmt.Sprintf("--workspace_status_command=./build/bazelutil/stamp.sh %s %s release", targetTriple, opts.Channel))
+		buildArgs = append(buildArgs, fmt.Sprintf("--workspace_status_command=./build/bazelutil/stamp.sh %s %s release %s", targetTriple, opts.Channel, opts.BuildTag))
```

Comment:
Why not just add a check here in this function, like `assertEq(opts.BuildTag, contentsOf("pkg/build/version.txt"))`? Doesn't that accomplish the same stated goal without complicating the existing logic any further?

There is also *already* a place in this same file where we call `cockroach version`, but we don't parse the results or anything. Would be trivial to add a check to parse out the `Build Tag` from that and make sure it is the same as `opts.BuildTag`. That is probably the most simple way to accomplish the stated goal.

Reply:
That assertion isn't true for continuous releases (nightlies, whatever you want to call them) is it?

Reply:
That's right, the assertion is only accurate for actual releases. We will probably update the pipeline to split these continuous releases into a new "category" in the [way I described in my comment over here](https://github.com/cockroachdb/cockroach/issues/100532#issuecomment-1496244770), so these binaries will not have build type `release` but something else (for example `cd`).

Reply:
yep, sounds good. I have the assertion started over in https://github.com/cockroachdb/cockroach/pull/100745 and will adapt this change to just (a) configure different "release" build types so that info.go can compose correct pre-release version strings for "nightly" / "continuous" / "preview" release builds and then (b) change how build_name is picked in the CI scripts to so that it uses the same format.